pkg/helm/engine/ownerref_test.go

```diff
@@ -31,53 +37,36 @@ func (e *mockEngine) Render(chrt *chart.Chart, v chartutil.Values) (map[string]s
 	return e.out, nil
 }
 
-func TestOwnerRefEngine(t *testing.T) {
-	ownerRefs := []metav1.OwnerReference{
-		{
-			APIVersion: "v1",
-			Kind:       "Test",
-			Name:       "test",
-			UID:        "123",
-		},
-	}
-
-	baseOut := `apiVersion: stable.nicolerenee.io/v1
-kind: Character
-metadata:
-  name: nemo
-spec:
-  Name: Nemo
-`
+type testCase struct {
+	input    map[string]string
+	expected map[string]string
+}
 
-	expectedOut := `---
-apiVersion: stable.nicolerenee.io/v1
-kind: Character
-metadata:
-  name: nemo
-  ownerReferences:
-  - apiVersion: v1
-    kind: Test
-    name: test
-    uid: "123"
-spec:
-  Name: Nemo
-`
-	expected := map[string]string{"template.yaml": expectedOut, "template2.yaml": expectedOut}
+type restMapping struct {
+	gvk   schema.GroupVersionKind
+	scope meta.RESTScope
+}
 
-	baseEngineOutput := map[string]string{
-		"template.yaml":  baseOut,
-		"template2.yaml": baseOut,
-		"empty.yaml":     "",
-		"comment.yaml":   "# This is empty",
+func genTemplate(resourceCount int, withLeadingSep bool, gvk schema.GroupVersionKind, ownerRefs []metav1.OwnerReference) string {
+	sb := &strings.Builder{}
+
+	for i := 0; i < resourceCount; i++ {
+		if withLeadingSep || i > 0 {
+			sb.WriteString("---\n")
```

Comment:
Nothing to fix, just a passing comment.
https://github.com/golang/go/blob/3a1b4e75f8b6c1b57db73bccf7ca871bf1a97ca9/src/strings/builder.go#L118-L120
```
 // WriteString appends the contents of s to b's buffer.
// It returns the length of s and a nil error.
```

Why return an error in the signature then? I suspect it probably did return an error at one point before they fixed it and decided not break the signature.

Reply:
I bet that it's actually to meet a `StringWriter` interface where other implementations actually do return errors.